packages/authorization-platformstate-writer/src/consumerServiceV2.ts

```diff
@@ -412,6 +414,64 @@ export async function handleMessageV2(
         })
       );
     })
+    .with({ type: "ClientAdminSet" }, async (msg) => {
+      const client = parseClient(msg.data.client, msg.type);
+      const pk = makePlatformStatesClientPK(client.id);
+      const clientEntry = await readPlatformClientEntry(pk, dynamoDBClient);
+
+      if (clientEntry && clientEntry.version > msg.version) {
+        logger.info(
+          `Skipping processing of entry ${clientEntry.PK}. Reason: a more recent entry already exists`
+        );
+        return Promise.resolve();
+      }
+
+      // platform-states
+      const platformClientEntry: PlatformStatesClientEntry = {
+        PK: pk,
+        state: itemState.active,
+        clientKind: clientKindToTokenGenerationStatesClientKind(client.kind),
+        clientConsumerId: client.consumerId,
+        clientPurposesIds: [],
+        version: msg.version,
+        updatedAt: new Date().toISOString(),
+        clientAdminId: client.adminId,
+      };
+      await upsertPlatformClientEntry(
+        platformClientEntry,
+        dynamoDBClient,
+        logger
+      );
+
+      // token-generation-states
+      await Promise.all(
+        client.keys.map(async (key) => {
+          const tokenGenStatesApiClient: TokenGenerationStatesApiClient = {
+            PK: makeTokenGenerationStatesClientKidPK({
+              clientId: client.id,
+              kid: key.kid,
+            }),
+            consumerId: client.consumerId,
+            clientKind: clientKindTokenGenStates.api,
+            publicKey: key.encodedPem,
+            GSIPK_clientId: client.id,
+            GSIPK_clientId_kid: makeGSIPKClientIdKid({
+              clientId: client.id,
+              kid: key.kid,
+            }),
+            updatedAt: new Date().toISOString(),
+            adminId: unsafeBrandId<UserId>(msg.data.adminId),
```

Comment:
```suggestion
            adminId: client.adminId,
```
My bad, this should be like this, because `client` is the new one that comes with the event meanwhile `clientEntry` is the old one, so we can correctly get `adminId` from `client`